src/sciline/pipeline.py

```diff
@@ -194,3 +199,93 @@ def bind_and_call(
     def _repr_html_(self) -> str:
         nodes = ((key, data) for key, data in self._graph.nodes.items())
         return pipeline_html_repr(nodes)
+
+
+def get_mapped_node_names(
+    graph: Pipeline, key: type, index_names: Sequence[Hashable] | None = None
+) -> pandas.Series:
+    """
+    Given a graph with key depending on mapped nodes, return a series corresponding
+    mapped keys.
+
+    This is meant to be used in combination with :py:func:`Pipeline.map`.
+    If the key depends on multiple indices, the series will be a multi-index series.
+
+    Note that Pandas is not a dependency of Sciline and must be installed separately.
+
+    Parameters
+    ----------
+    graph:
+        The pipeline to get the mapped key names from.
+    key:
+        The key to get the mapped key names for. This key must depend on mapped nodes.
+    index_names:
+        Specifies the names of the indices of the mapped node. If not given this is
+        inferred from the graph, but the argument may be required to disambiguate
+        multiple mapped nodes with the same name.
+
+    Returns
+    -------
+    :
+        The series of mapped key names.
+    """
+    import pandas as pd
+    from cyclebane.graph import IndexValues, MappedNode, NodeName
+
+    candidates = [
+        node
+        for node in graph._cbgraph.graph.nodes
```

Comment:
I don't like that this uses a protected attribute of `graph`. Should `get_mapped_node_names` be a method of `Pipeline`?

Reply:
This is definitely something that we need to consider for a lot of other functionality (in particular around planner graph operations) that we intend to implement. Either we need to make those properties accessible on the public interface, or potentially add a lot more methods to `Pipeline`. Right now I cannot say which is better? So unless you can clearly say which solution should be chosen, I'd like to keep this as it is for now.

Reply:
I don't know what operations you have in mind that need direct access to the graph. But I would say that we should expose a minimal set of primitive operations that can be composed in, e.g., methods of workflow classes.